### DISCUSSION TITLE
Pass basket into FPMC instead of individual items?

### DISCUSSION THREAD

**Discussion by gcbrown** [RUCAIBox/RecBole#1317] [Q&A] [answered]
Hello, I'm trying to generate some next-basket predictions. I have a dataset with columns (user, timestamp, basket), where basket is a set of ~10 items. I'd like to use FPMC to look at baskets as discrete sets like in the original paper. But in RecBole I can only figure out how to pass in individual items, which means I'm having to split the orders up into separate interactions/rows and I'm losing the discrete boundaries between orders. I can see that item_id_list is a potential input but I can't figure out how to pass it in, because whenever I put just item_id_list, run_recbole fails looking for the item_id column.

Is there something I need to pass into config_dict to tell the model to operate on token_seqs instead of individual tokens? Is it possible in RecBole?

Thanks

**Comment by Sherry-XLL** [comment, marked as answer]
@gcbrown Hello, `item_id_list` and `item_id` are generated automatically based on users' interactions in `data_augmentation` as follows. `item_id_list` is the historical item set in chronological order and `item_id` is the user's next interactive ground truth item.

https://github.com/RUCAIBox/RecBole/blob/ddff6211d030906f502a073a9f2b9e2250e93664/recbole/data/dataset/sequential_dataset.py#L73-L89

In RecBole, the existing code can only recommend individual items instead of basket items for the time being. If you want to operate on token_seqs instead of individual tokens, you should modify `SequentialDataset` in data preprocessing and refactor data loading according to your needs. Only changing the configuration file cannot realize the next-basket recommendation.

Thanks for your attention to RecBole.
